packages/design/src/DataTable/useTable.ts

```diff
@@ -39,13 +40,32 @@ export default function useTable<T>({
     };
   });
 
+  const [searchAndFilterCb] = useState<MatchCallback<T>>(() => {
+    return (
+      targetValue: any,
+      searchValue: string,
+      propName: keyof T & string
+    ) => {
+      for (const matcher of customSearchMatchers) {
+        const isMatched = matcher(targetValue, searchValue, propName);
+        if (isMatched) {
+          return true;
+        }
+      }
+
+      // No match found.
+      return false;
+    };
+  });
+
```

Comment:
lol i know i wrote this, but it's unnecessary pre-optimization (define the function once) lets define it regularly

```
function searchAndFilterCb(targetValue: any, searchValue: string, propName: keyof T & string) {
     for (const matcher of customSearchMatchers) {
        const isMatched = matcher(targetValue, searchValue, propName);
        if (isMatched) {
          return true;
        }
      }

      // No match found.
      return false;
}
```